crates/accelerate/src/unitary_synthesis.rs

```diff
@@ -0,0 +1,1093 @@
+    let mut available_2q_basis: IndexMap<&str, NormalOperation> = IndexMap::new();
+    let mut available_2q_props: IndexMap<&str, (Option<f64>, Option<f64>)> = IndexMap::new();
+
+    let mut qubit_gate_map = IndexMap::new();
+    match target.operation_names_for_qargs(Some(&qubits)) {
+        Ok(direct_keys) => {
+            qubit_gate_map.insert(&qubits, direct_keys);
+            if let Ok(reverse_keys) = target.operation_names_for_qargs(Some(&reverse_qubits)) {
+                qubit_gate_map.insert(&reverse_qubits, reverse_keys);
+            }
+        }
+        Err(_) => {
+            if let Ok(reverse_keys) = target.operation_names_for_qargs(Some(&reverse_qubits)) {
+                qubit_gate_map.insert(&reverse_qubits, reverse_keys);
+            } else {
+                return Err(QiskitError::new_err(
+                    "Target has no gates available on qubits to synthesize over.",
+                ));
+            }
+        }
+    }
+
+    #[inline]
+    fn replace_parametrized_gate(mut op: NormalOperation) -> NormalOperation {
+        if let Some(std_gate) = op.operation.try_standard_gate() {
+            match std_gate.name() {
+                "rxx" => {
+                    if let Param::ParameterExpression(_) = op.params[0] {
+                        op.params[0] = Param::Float(PI2)
+                    }
+                }
+                "rzx" => {
+                    if let Param::ParameterExpression(_) = op.params[0] {
+                        op.params[0] = Param::Float(PI4)
+                    }
+                }
+                "rzz" => {
+                    if let Param::ParameterExpression(_) = op.params[0] {
+                        op.params[0] = Param::Float(PI2)
+                    }
+                }
+                _ => (),
+            }
+        }
+        op
+    }
+
+    for (q_pair, gates) in qubit_gate_map {
+        for key in gates {
+            match target.operation_from_name(key) {
+                Ok(op) => {
+                    match op.operation.view() {
+                        OperationRef::Gate(_) => (),
+                        OperationRef::Standard(_) => (),
+                        _ => continue,
+                    }
+
+                    available_2q_basis.insert(key, replace_parametrized_gate(op.clone()));
+
+                    if target.qargs_for_operation_name(key).is_ok() {
+                        available_2q_props.insert(
+                            key,
+                            match &target[key].get(Some(q_pair)) {
+                                Some(Some(props)) => (props.duration, props.error),
+                                _ => (None, None),
+                            },
+                        );
+                    } else {
+                        continue;
+                    }
+                }
+                _ => continue,
+            }
+        }
+    }
```

Comment:
Since you're only handling the `Ok()` cases here, you could rewrite this more neatly by using `flatten()`.
```suggestion
    for (q_pair, gates) in qubit_gate_map {
        for op in gates.iter().map(|key|target.operation_from_name(*key)).flatten() {
            match op.operation.view() {
                OperationRef::Gate(_) => (),
                OperationRef::Standard(_) => (),
                _ => continue,
            }

            available_2q_basis.insert(key, replace_parametrized_gate(op.clone()));

            if target.qargs_for_operation_name(key).is_ok() {
                available_2q_props.insert(
                    key,
                    match &target[key].get(Some(q_pair)) {
                        Some(Some(props)) => (props.duration, props.error),
                        _ => (None, None),
                    },
                );
            } else {
                continue;
            }
        }
    }
```